src/cmd/src/metasrv.rs

```diff
@@ -251,4 +238,66 @@ mod tests {
         assert_eq!("/tmp/greptimedb/test/logs", logging_opt.dir);
         assert_eq!("debug", logging_opt.level);
     }
+
+    #[test]
+    fn test_config_precedence_order() {
+        let mut file = create_named_temp_file();
+        let toml_str = r#"
+            bind_addr = "127.0.0.1:3002"
+            server_addr = "127.0.0.1:3002"
+            datanode_lease_secs = 15
+            selector = "LeaseBased"
+            use_memory_store = false
+
+            [logging]
+            level = "debug"
+            dir = "/tmp/greptimedb/test/logs"
+        "#;
+        write!(file, "{}", toml_str).unwrap();
+
+        let env_vars_prefix = "METASRV_UT";
+        temp_env::with_vars(
+            vec![
+                (
+                    // bind_addr = 127.0.0.1:14002
+                    vec![env_vars_prefix.to_string(), "bind_addr".to_uppercase()].join(ENV_VAR_SEP),
+                    Some("127.0.0.1:14002"),
+                ),
+                (
+                    // server_addr = 127.0.0.1:13002
+                    vec![env_vars_prefix.to_string(), "server_addr".to_uppercase()]
+                        .join(ENV_VAR_SEP),
+                    Some("127.0.0.1:13002"),
+                ),
+            ],
+            || {
+                let command = StartCommand {
+                    bind_addr: None,
+                    server_addr: Some("127.0.0.1:23002".to_string()),
+                    store_addr: None,
+                    selector: None,
+                    config_file: Some(file.path().to_str().unwrap().to_string()),
+                    use_memory_store: false,
+                    http_addr: None,
+                    http_timeout: None,
+                    env_vars_prefix: env_vars_prefix.to_string(),
```

Comment:
suggestion: Consider using `..Default::default()`?

Reply:
you're right... I forget it.